crates/common/checkpoint_sync/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+pub mod checkpoint;
+use alloy_primitives::B256;
+use anyhow::ensure;
+use checkpoint::get_checkpoint_sync_sources;
+use ream_consensus::{
+    blob_sidecar::{BlobIdentifier, BlobSidecar},
+    constants::SECONDS_PER_SLOT,
+    electra::{beacon_block::SignedBeaconBlock, beacon_state::BeaconState},
+    execution_engine::rpc_types::get_blobs::BlobAndProofV1,
+};
+use ream_fork_choice::{handlers::on_tick, store::get_forkchoice_store};
+use ream_network_spec::networks::network_spec;
+use ream_rpc::types::response::BeaconVersionedResponse;
+use ream_storage::{db::ReamDB, tables::Table};
+use reqwest::Url;
+use tracing::{info, warn};
+
+/// Entry point for checkpoint sync.
+pub async fn initialize_db_from_checkpoint(
+    db: ReamDB,
+    checkpoint_sync_url: Option<Url>,
+) -> anyhow::Result<()> {
+    if db.is_initialized() {
+        warn!("Starting Checkpoint Sync from existing DB. It is advised to start from a fresh DB");
+        return Ok(());
+    }
+
+    let checkpoint_sync_url = get_checkpoint_sync_sources(checkpoint_sync_url)[0].clone();
+    info!("Starting Checkpoint Sync");
+    let block = fetch_finalized_block(&checkpoint_sync_url).await?;
+    info!("Fetched Block Root:{}", block.data.message.block_root());
+    let slot = block.data.message.slot;
+    initialize_blobs_in_db(
+        &checkpoint_sync_url,
+        db.clone(),
+        block.data.message.block_root(),
+    )
+    .await?;
+    info!("Received block from slot:{}", slot);
+
+    let state = get_state(&checkpoint_sync_url, block.data.message.slot).await?;
+    info!("Fetched State Root:{}", state.data.state_root());
+    ensure!(
+        block.data.message.slot == state.data.slot,
+        anyhow::anyhow!("Slot mismatch")
```

Comment:
```suggestion
        "Slot mismatch"
```

can't we just do this?